src/BlackLinter.php

```diff
@@ -53,42 +52,24 @@ public function getVersion() {
     return trim(str_replace('black, version', '', $stdout));
   }
 
-  public function getLinterConfigurationOptions() {
-    $options = array(
-      'black.linelen' => array(
-        'type' => 'optional int',
-        'help' => pht('Specify a non-default line length for black'),
-      ),
-    );
-    return $options + parent::getLinterConfigurationOptions();
-  }
-
-  public function setLinterConfigurationValue($key, $value) {
-    switch ($key) {
-      case 'black.linelen':
-        $this->linelen = $value;
-        return;
-    }
-    return parent::setLinterConfigurationValue($key, $value);
-  }
-
-  protected function getMandatoryFlags() {
-    $flags = array('--quiet');
-    if ($this->linelen != null) {
-        array_push($flags, '--line-length', $this->linelen);
-    }
-    return $flags;
-  }
-
   public function getInstallInstructions() {
     return pht('pip3 install black');
   }
 
   protected function parseLinterOutput($path, $err, $stdout, $stderr) {
     if ($err == 123 or $stderr) {
       return false;
-    } else {
-      return array();
     }
+    $flags = $this->getCommandFlags();
+    // Remove --check flag since instructions to user should be to fix lint errors 
+    unset($flags[array_search('--check', $flags)]);
+
+    $message = new ArcanistLintMessage();
+    $message->setPath($path);
+    $message->setName($this->getLinterName());
+    $message->setDescription("Please run `".$this->getLinterConfigurationName()." ".join(" ", $flags)." ".$path."`\n");
```

Comment:
Let's just go back to using `black` as a literal in this string instead. We're not gaining anything by using `getLinterConfigurationName()`.